File: utils/Optimization/generate_path_with_time.m
function [path_with_time timelist] = generate_path_with_time(path,traj_constant,T_seg,t_start)

if nargin == 4
    
    time = t_start;
    time_list = [time];
    traj_num = length(T_seg);
    
    for j = 1:traj_num
        time = time+T_seg(j);
        time_list = [time_list;time];
    end
    
    path_with_time = [path time_list];
    timelist = path_with_time(:,4);
    
elseif nargin == 3
    
    time = 0.0;
    time_list = [time];
    %traj_num = length(T_seg);
    
    for j = 1:traj_constant.total_traj_num
        time = time+T_seg(j);
        time_list = [time_list;time];
    end
    
    path_with_time = [path time_list];
    timelist = path_with_time(:,4);
    
else
    error('The number of input arguments are not correct');
end



end

File: rotor_tm_traj/traj/Optimization/generate_path_with_time.py
#! /usr/bin/env python

import numpy as np
# test passed
def generate_path_with_time(path=None, traj_constant=None, T_seg=None, t_start=None):
    if (np.any(path!= None) ) and (np.any(traj_constant != None)) and (np.any(T_seg != None)) and (np.any(t_start != None)):
        time = t_start
        time_list = np.array([[time]])
        traj_num = T_seg.shape[0] # may be shape[1]

        for j in range(1, traj_num+1):
            time = time + T_seg[j-1]
            time_array = np.array([[time]])
            time_list = np.append(time_list, time_array, axis=0)        
            
        path_with_time = np.append(path, time_list, axis=1)
        timelist = path_with_time[:, 3]

    elif (np.any(path!= None) ) and (np.any(traj_constant != None)) and (np.any(T_seg != None)):
        time = 0.0
        time_list = np.array([[time]])

        for j in range(1, traj_constant.total_traj_num+1):
            time = time + T_seg[j-1]
            time_array = np.array([[time]])
            time_list = np.append(time_list, time_array, axis=0)

        path_with_time = np.append(path, time_list, axis=1)
        timelist = path_with_time[:, 3:]

    return path_with_time,timelist